Source/Initialization/PlasmaInjector.cpp

```diff
@@ -497,8 +505,12 @@ XDim3 PlasmaInjector::getMomentum (Real x, Real y, Real z) const noexcept
 bool PlasmaInjector::insideBounds (Real x, Real y, Real z) const noexcept
 {
     return (x < xmax and x >= xmin and
+#ifndef WARPX_DIM_3D
```

Comment:
this will disable the y bounds check for particles in RZ. A better check might be ([docs](https://warpx.readthedocs.io/en/latest/developers/dimensionality.html)):
```suggestion
#ifdef WARPX_DIM_XZ
```

Not using the `y` parameter will cause a compile-time warning in 2D. We can add `(void)y;` before the return line to suppress it.

Potentially cleaner is to not modify the `insideBounds` function at all and instead keep in `WARPX_DIM_XZ` the `ymin`/`ymax` with `std::numeric_limits<amrex::Real>::lowest()`/`std::numeric_limits<amrex::Real>::max()` instead, see note above.

Reply:
I don't understand what you wrote here.
I am defining `y` and `uy` both. And I am using both in the `injectBounds()` function.
Also I tested in 2D and got no warning, you can see the output in here:
[output.txt](https://github.com/ECP-WarpX/WarpX/files/4794569/output.txt)

Reply:
Sorry, at compile time I don't remember seeing any warnings.
But I will check it again after confirming that plasmas initialized with moving window and outside of the box are injected correctly.

Reply:
From the current version, I see no warnings at compile time.